lib/dal.js

```diff
@@ -12,26 +29,32 @@ var customError = require('./customError.js');
 var getSharedExtension = function(confFile, name) {
   var defer = Q.defer();
   var sharedExtension = {};
-
   sharedExtension.found = false;
-  return fetchJSON(confFile).then(function(data) {
-    data.sharedExtensions.forEach(function(extension, i) {
+  fetchJSON(confFile).then(function(data) {
```

Comment:
There's no reason to have a deferred here. fetchJSON already is async. Just return the result of that. Everywhere where you reject, just throw instead, and everywhere where you resolve return instead.